### DISCUSSION TITLE
How can i connect Pocketbase to SPA

### DISCUSSION THREAD

**Discussion by MuslemRahimi** [pocketbase/pocketbase#3462] [Q&A] [answered]
Hi,

In my sveltekit app i currently have in my hooks.server.ts:
`
import PocketBase from 'pocketbase';
import { serializeNonPOJOs } from '$lib/utils';

export const handle = async ({ event, resolve }) => {

	event.locals.pb = new PocketBase(import.meta.env.VITE_POCKETBASE_URL);
	event.locals.pb.authStore.loadFromCookie(event?.request?.headers?.get('cookie') || '');

	try {
		if (event.locals.pb.authStore.isValid) {
			await event.locals.pb.collection('users').authRefresh();
			event.locals.user = serializeNonPOJOs(event.locals.pb.authStore.model);
		}
	} catch(_) {
		event.locals.pb.authStore?.clear();
		event.locals.user = undefined;
	}

	

	

	const response = await resolve(event);

	response.headers.set('set-cookie', event?.locals?.pb?.authStore?.exportToCookie({ sameSite: 'lax', secure: false }));

	return response;
};
`
Is there a way to rewrite this code so that Pocketbase is connected on the client side?
Any help is much appreciated

**Comment by ganigeorgiev** [comment, marked as answer]
The question is too vague and I'm not sure that I understand what you are trying to do.

When you say _"rewrite this code so that Pocketbase is connected on the client side"_ how exactly are you planning your SvelteKit app to to interact with PocketBase?

Since you mentioned SPA do you mean that you use the SvelteKit static adapter, aka. your application is entirely client-side?

If that's the case you don't have to do anything special. Just create an instance of the PocketBase SDK and import it in your components (it is safe to have a single global instance on the client-side).

**But** if you want a "mixed access", aka. to maintain your users state both in the browser and node sides, then you'll need:

1. Export from the server-side the cookie with `httpOnly: false` to allow JS access to it:

    ```js
    event.locals.pb.authStore.exportToCookie({ httpOnly: false }))
    ```

2. Load and set the cookie on the client-side (eg. with `document.cookie`):
    ```js
    // src/pb.js

    import PocketBase from 'pocketbase'

    const pb = new PocketBase('https://example.com')

    // the below 2 calls are only needed if you want to maintain the auth state in both browser and node
    pb.authStore.loadFromCookie(document.cookie)
    pb.authStore.onChange(() => {
        document.cookie = pb.authStore.exportToCookie({ httpOnly: false })
    })

    export default pb
    
    // then in your components you can just import it and use it directly
    // import pb from "./pb.js"
    ```

**Reply by MuslemRahimi** [reply]
thank you for your answer
